Grunt Added

metaColin opened the pull request:
Hey Dan,

Codekit is great, but I've outgrown it.

Grunt.js is the heavy duty taskrunner, for folks who need a little more elbow grease than Codekit has. I created the gruntfiles necessary for brew. Please consider adding them into the master.

Added Grunt functionality for:
-js concatenation and minification
-image optimization
-css to less

metaColin committed:
Grunt Added

Added Grunt functionality for:
-js concatenation and minification
-image optimization
-css to less

slightlyoffbeat commented:
Thank you for this.  I will review.  Grunt was one thing I wanted to make sure was implemented for 1.0

metaColin committed:
Disabled grunt imagemin

imagemin is not stable, it can nuke your project images.

metaColin commented:
I had to disable the imagemin function in the gruntfile. Turns out it's not stable and sometimes instead of optimizing images, will instead nuke them.

kLOsk commented:
Any news on approving this for master?

metaColin committed:
ignore grunt dependencies folder

gitignore fixed to ignore /node_modules
it was sloppy of me to commit the dependencies folder in the first
place.

metaColin committed:
Removed Grunt Dependencies

When you run grunt for the first time these dependencies are
downloaded, no need to include them in the project files.

metaColin commented:
I updated the .gitignore to ignore the /node_modules folder. That folder is all grunt dependencies that get downloaded the first time you run grunt. No reason for it to be included in the git repo.

slightlyoffbeat commented:
Hello!  Soon :) I've been away on business.  I'm back and plan to start reviewing pull requests and pushing on towards 1.0. Thanks!

Dan

metaColin commented:
Not sure where to put this, maybe in the read me. For those not familiar with grunt, make sure you install it, I suggest this guide: http://www.ghosthorses.co.uk/production-diary/installing-grunt-on-os-x-and-windows-7/
Once installed a few steps to set it up for brew:
1) from your terminal navigate to the brew theme directory
2) run the command 'npm install' (this will download grunt dependencies)
3) run a grunt command:
    -'grunt' to compile once
    -'grunt watch' to watch for changes and compile on each save

metaColin committed:
Fixed Error

call_user_func() expects parameter 1 to be a valid callback

I don’t know what caused this. So I grabbed a fresh version of brew and
re-added:
-Gruntfile.js
-package.json
-updated .gitignore

metaColin committed:
Missed Global

Tanmccuin commented:
@colinsteinmann or @slightlyoffbeat - How would i go about grabbing these files so i run grunt on my vagrant box?

metaColin commented:
Hey Tanner,

Are you just asking how to grab the fork of the brew theme with grunt files included?
The un-merged fork is located here: https://github.com/colinsteinmann/brew

Or are you asking something specific about using these files with Vagrant?
The procedure for setting up Grunt and the procedure for installing a new Wordpress theme (such as brew) are not changed by using Vagrant. It's no different from running Ampps, or Xampp, or whatever dev environment:

1) Setup up a new Wordpress install, and put the directory folder in your theme directory.

2) Make sure Grunt is installed locally (not on your Vagrant VM). See the link to instructions in an earlier thread comment I made.

3) Open a terminal and cd to the brew theme directory, then type in 'grunt watch'.

P.S. The grunt fork is no doubt a little bit behind the master fork at this point, so be aware of this tradeoff.

aadesh commented:
Hey Slightlyoffbeat,

Any idea when you plan to merge the pull request?
- Aadesh
